File: frontend/src/types/recommendations.ts
export interface Instrument {
  symbol: string;
  name: string;
  type: 'stock' | 'ETF';
  reason: string;
  risk: string;
  selection_score?: number;
  role?: string;
  factors?: Record<string, number>;
  why_selected?: string[];
  risk_notes?: string[];
}

export interface AllocationSuggestion {
  aggressive: string;
  moderate: string;
  conservative: string;
}

export interface RejectedAlternative {
  symbol: string;
  name: string;
  selection_score: number;
  reason_rejected: string;
}

export interface SectorRecommendation {
  sector: string;
  direction: 'bullish' | 'neutral' | 'bearish';
  confidence: number;
  time_horizon: 'short_term' | 'medium_term' | 'long_term';
  recommended_instruments: Instrument[];
  allocation_suggestion: AllocationSuggestion;
  why_now: string;
  key_drivers: string[];
  risks: string[];
  summary: string;
  profile_fit?: {
    conservative: string[];
    moderate: string[];
    aggressive: string[];
  };
  rejected_alternatives?: RejectedAlternative[];
  strategy_note?: string;
}

export interface PortfolioAllocation {
  sector: string;
  weight_pct: number;
  instrument: string;
  rationale: string;
}

export interface PortfolioStrategy {
  profile: 'aggressive' | 'moderate' | 'conservative';
  allocations: PortfolioAllocation[];
  cash_pct: number;
  summary: string;
}

export interface RecommendationsResponse {
  generated_at: string;
  market_context: string;
  top_recommendations: SectorRecommendation[];
  portfolio_strategy: PortfolioStrategy[];
  disclaimer: string;
}

File: backend/app/schemas/recommendations.py
from pydantic import BaseModel
from typing import Literal
from datetime import datetime


class Instrument(BaseModel):
    symbol: str
    name: str
    type: Literal["stock", "ETF"]
    reason: str
    risk: str


class AllocationSuggestion(BaseModel):
    aggressive: str
    moderate: str
    conservative: str


class SectorRecommendation(BaseModel):
    sector: str
    direction: Literal["bullish", "neutral", "bearish"]
    confidence: int
    time_horizon: Literal["short_term", "medium_term", "long_term"]
    recommended_instruments: list[Instrument]
    allocation_suggestion: AllocationSuggestion
    why_now: str
    key_drivers: list[str]
    risks: list[str]
    summary: str


class PortfolioAllocation(BaseModel):
    sector: str
    weight_pct: int
    instrument: str
    rationale: str


class PortfolioStrategy(BaseModel):
    profile: Literal["aggressive", "moderate", "conservative"]
    allocations: list[PortfolioAllocation]
    cash_pct: int
    summary: str


class RecommendationsResponse(BaseModel):
    generated_at: datetime
    market_context: str
    top_recommendations: list[SectorRecommendation]
    portfolio_strategy: list[PortfolioStrategy]
    disclaimer: str
